### DISCUSSION TITLE
How to properly use update human user API?

### DISCUSSION THREAD

**Discussion by ashep** [zitadel/zitadel#8677] [Q&A] [answered]
Hello there!

I am trying to update a user through the [user update API](https://zitadel.com/docs/apis/resources/user_service_v2/user-service-update-human-user).

```sh
curl --request PUT \
  --url http://zitadel:8081/v2/users/human/82e7cbc9-1d8e-40d1-8b1b-47c719efec43 \
  --header 'Authorization: Bearer TOKEN' \
  --header 'Content-Type: application/json' \
  --data '{
	"profile": {
		"givenName": "Minnie",
		"familyName": "Mouse",
		"displayName": "Minnie Mouse"
	}
}
'
```

I receive the 200 code in response with the following response:

```json
{
	"details": {
		"sequence": "1",
		"changeDate": "2024-09-24T09:48:58.866069Z",
		"resourceOwner": "286345700260708354"
	}
}
```

But the user isn't updated. What am I doing wrong?

Zitadel version is 2.58.0.

Many thanks for considering my request.

**Comment by fforootd** [comment]
Hm that looks odd indeed.

What happens if you use `--data-raw` instead `--data`?

**Reply by ashep** [reply]
> What happens if you use --data-raw instead --data?

Nothing changes. Actually, it doesn't matter what HTTP client to use for the request. I've provided a cURL version just as a demo.

![image](https://github.com/user-attachments/assets/f8fa97e7-e1f8-4be7-a662-72d06c5a78be)

**Reply by ashep** [reply]
I am wondering whether the request body is correct.

**Reply by fforootd** [reply]
Hm @hifabienne I am not aware that this is also wrong in the docs, or do you know something on this call?

**Reply by hifabienne** [reply]
I can reproduce it, but couldn't figure out the problem yet. 
We are looking into it and get back to you

**Reply by stebenz** [reply, marked as answer]
@ashep we have a fix for that, will include it in the next release, refer to this PR for more information https://github.com/zitadel/zitadel/pull/8635

**Comment by mTpChris** [comment]
I think you have to provide the full data object object as it is a PUT method route. And the response shouldn't be 200 then.
It would be nice to have a PATCH method route to do this. 
We really need this as it make sense for a service user to only update some part of the profile without knowing a password or even setting a new one.
Thanks.

Edit:
I've just see v2.63.0 solve this issue using the PUT route. It should still be a PATCH method but it's ok it works now. 
Thanks for this update.
